src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/CSSISpaceWeather.cpp

```diff
@@ -0,0 +1,588 @@
+            };
+
+            spaceWeather.monthlyPredictions_.insert({mjd, reading});
+        }
+    }
+
+    if (!spaceWeather.observations_.empty())
+    {
+        spaceWeather.lastObservationDate_ = spaceWeather.observations_.rbegin()->second.date;
+
+        const Instant observationStartInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.observations_.begin()->first), Scale::UTC);
+        const Instant observationEndInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.observations_.rbegin()->first), Scale::UTC);
+
+        spaceWeather.observationInterval_ = Interval::Closed(observationStartInstant, observationEndInstant);
+    }
+
+    if (!spaceWeather.dailyPredictions_.empty())
+    {
+        const Instant dailyPredictionStartInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.dailyPredictions_.begin()->first), Scale::UTC);
+        const Instant dailyPredictionEndInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.dailyPredictions_.rbegin()->first), Scale::UTC);
+
+        spaceWeather.dailyPredictionInterval_ =
+            Interval::Closed(dailyPredictionStartInstant, dailyPredictionEndInstant);
+
+        // Use the last daily prediction to make an artificial first monthly prediction
+        // so that the data overlaps
+        const CSSISpaceWeather::DailyPrediction& lastDailyPrediction = spaceWeather.dailyPredictions_.rbegin()->second;
+
+        Date monthBeginningDate = lastDailyPrediction.date;
+        monthBeginningDate.setDay(1);
+
+        const CSSISpaceWeather::MonthlyPrediction overlapMonthlyReading = {
+            monthBeginningDate,
+            lastDailyPrediction.BSRN,
+            lastDailyPrediction.ND,
+            lastDailyPrediction.ISN,
+            lastDailyPrediction.F107_OBS,
+            lastDailyPrediction.F107_ADJ,
+            lastDailyPrediction.F107_DATA_TYPE,
+            lastDailyPrediction.F107_OBS_CENTER81,
+            lastDailyPrediction.F107_OBS_LAST81,
+            lastDailyPrediction.F107_ADJ_CENTER81,
+            lastDailyPrediction.F107_ADJ_LAST81,
+        };
+        const Integer monthMjd = DateTime(monthBeginningDate, Time::Midnight()).getModifiedJulianDate().floor();
+        spaceWeather.monthlyPredictions_.insert({monthMjd, overlapMonthlyReading});
+    }
+
+    if (!spaceWeather.monthlyPredictions_.empty())
+    {
+        const Instant monthlyPredictionStartInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.monthlyPredictions_.begin()->first), Scale::UTC);
+        const Instant monthlyPredictionEndInstant =
+            Instant::ModifiedJulianDate(Real::Integer(spaceWeather.monthlyPredictions_.rbegin()->first), Scale::UTC);
+
+        spaceWeather.monthlyPredictionInterval_ =
+            Interval::Closed(monthlyPredictionStartInstant, monthlyPredictionEndInstant);
+    }
+
+    return spaceWeather;
+}
+
+CSSISpaceWeather::CSSISpaceWeather()
```

Comment:
Why have a private default constructor at all? And not just use `CSSISpaceWeather::Undefined()` and define this over there?